csharp/src/Microsoft.ML.OnnxRuntime/ProviderOptions.shared.cs

```diff
@@ -218,6 +218,98 @@ protected override bool ReleaseHandle()
     }
 
 
+    /// <summary>
+    /// Holds the options for configuring a ROCm Execution Provider instance
+    /// </summary>
+    public class OrtROCMProviderOptions : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+
+        #region Constructor
+
+        /// <summary>
+        /// Constructs an empty OrtROCMroviderOptions instance
+        /// </summary>
+        public OrtROCMProviderOptions() : base(IntPtr.Zero, true)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateROCMProviderOptions(out handle));
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Get ROCm EP provider options
+        /// </summary>
+        /// <returns> return C# UTF-16 encoded string </returns>
+        public string GetOptions()
+        {
+            var allocator = OrtAllocator.DefaultInstance;
+
+            // Process provider options string
+            IntPtr providerOptions = IntPtr.Zero;
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtGetROCMProviderOptionsAsString(handle, allocator.Pointer, out providerOptions));
+            using (var ortAllocation = new OrtMemoryAllocation(allocator, providerOptions, 0))
+            {
+                return NativeOnnxValueHelper.StringFromNativeUtf8(providerOptions);
+            }
```

Comment:
Please, refer to this PR for important modifications on how we handle pinning and memory allocaitons.

Reply:
- [ ] follow the change in PR mentioned

Reply:
We hope to merge that PR soon.

Reply:
No problem. I will do a rebase once it is merged, and then update this one.